### ISSUE TITLE
`enum` being wrapped that isn't used by any wrapped functions

### ISSUE THREAD

**Issue by halleysfifthinc** [grasph/wrapit#16]
A class I'm wrapping has some functions that are only public when a given `#define` is present, but the functions are created in the .cpp file regardless. `wrapit` is (correctly) not wrapping those functions, but an enum that is only used (as an argument) by those functions is still being wrapped. I tried adding those functions to the veto header (just in case that affected type/enum wrapping, even though it wasn't necessary to prevent wrapping the functions themselves), but the enum was still being wrapped.

tl;dr: An enum is being wrapped that isn't used as a return or argument type for a wrapped function.

I have a related, but more meta level question: what is the intended automation level of `wrapit`? Is it supposed to be more like SWIG, where bindings are fully generated automatically, or is some level of manual correction after generation expected/intended/unavoidable? Either way, I'm really glad `wrapit` exists, because I have no C++ experience!

**Comment by grasph** [comment]
Helloe @halleysfifthinc ,

Can you provide a minimum example illustrating the issue with `enum` ?

Concerning the automation level, the aim is the generation of code that do not require any manual correction.

Philippe.
